### PR TITLE
Chess: builds out the board, peices, and basic logic

### PR DESCRIPTION
**Pull request by TrevorJamesH**
Todo: 
advanced game logic: checking collisions, pawn attack, castle, en passant, queening a pawn, and checkmate check

### CODE REVIEW

**Comment by Moniarchy** [comment]
@TrevorJamesH would you mind adding your specs to your README or to a CONTRACT.md so I know what you've already accomplished? Thank you. :)

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
+import React, { Component } from 'react'
+import gameLogic from './js/Logic.js'
+import {Pawn, Bishop, Rook, Queen, King, Knight} from './components/pieces'
+import './css/board.css'
+
+
+class Square{
```
Comment (Moniarchy) [inline src/App.js:7]: is this `Square` a regular class and not a React component for any particular reason?

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    let piece = this.state.board[y][x].piece
+    if( piece.color === this.state.nextPlayer){
+      this.setState({selected: {x:x,y:y}})
+      return
+    }
+    if(this.selectedSquare() && this.isLegal(x,y)){
+      console.log(this.state.selected,' clicked')
+      this.movePiece(x,y)
+    }
+    else if(piece != 'Blank' && piece.color === this.state.nextPlayer){
```
Comment (Moniarchy) [inline src/App.js:73]: You might want this on line 72 just for human readability--just so I know that this logic is the antithesis of the condition starting on 69.

File: src/App.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+    return(
+      <div className="Board">
+        {boardRender}
+      </div>
+    )
+  }
+}
+
+export default Game
```
Comment (Moniarchy) [inline src/App.js:131]: This module looks great, Trevor.

File: src/components/pieces.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
+class Pawn{
+  constructor( color ){
+    this.color = color
+    this.name = 'Pawn'
+    this.moves = 0
```
Comment (Moniarchy) [inline src/components/pieces.js:5]: Out of curiosity (seeking a response to this question for my own learning), why are we keeping track of the number of moves a piece has?

File: src/css/board.css [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+.WhiteSquare{
+  background-color: white;
+  color: black;
+}
+
+.Highlight{
+  box-shadow:0px 0px 0px 4px red inset;
+}
+
+.White.King:after{
```
Comment (Moniarchy) [inline src/css/board.css:33]: Potential refactor:

```
.White.Queen:after,
.White.Rook:after,
.White.Bishop:after,
...all the way down to 
.Black.Pawn:after
 { font-size:40px }
```

**Review by Moniarchy — APPROVED** [review]
This looks so wonderful. I dropped some small notes in, but this is totally ready to be merged.

File: src/components/pieces.js (hunk 4 above)
Comment (TrevorJamesH) [reply src/components/pieces.js:5]: Because pawns can move differently on their first move.

**Pull request merged**